src/gui/mzroll/spectrawidget.cpp

```diff
@@ -236,6 +236,11 @@ void SpectraWidget::overlayCompoundFragmentation(Compound* c)
 	
     SpectralHit hit;
   	hit.score = 0;
+    //TODO: precursormz should be preset as the compound m/z
+    //compound->mass should be reserved for exact mass or renamed
+    if (!c->formula.empty())
+        c->precursorMz = c->adjustedMass(mainwindow->mavenParameters->getCharge(c));
+    if (c->precursorMz == 0) c->precursorMz = c->mass;
```

Comment:
Please use the new `almostEqual` method to compare two floating point (or double) values:
```C++
if (mzUtils::almostEqual(c->precursorMz, 0.0f))
    c->precursorMz = c->mass;
```